jccli/cli.py

```diff
@@ -31,14 +33,19 @@ def abort_if_false(ctx, param, value):
 
 
 @click.group()
-@click.option('--key', "-k", required=True, type=str, help='Jumpcloud API key')
+@click.option('--key', "-k", type=str, help='Jumpcloud API key (can also use environmental variable: JCCLI_API_KEY)', envvar='JCCLI_API_KEY')
+@click.option('--key-file', '-K', type=str, help='Path to text file containing Jumpcloud API key')
```

Comment:
i think we need some documentation for this?    What should the entries look like?  

Also instead of passing in the path to a file could we just have a default configurations file in user home?  something like a `~/.jccli` file that contains configurations for the CLI?

Reply:
I was just imagining a file whose only contents are the key itself.

What would be in the `.jccli` file, other than the API key?

Reply:
well i can image that we could put the logger configuration in there.  also lets say we create an output formatter for our GET requests, like `--output [text, yaml, json]`.  We could put the default ouput configuration in there as well.

Reply:
@zaro0508 I think regardless of how we implement the `~/.jccli` config file, we will want users to be able to store their keys outside of the `~/.jccli` file. I'm thinking of this because a lot of people like to track their dotfiles in version control, which they obviously can't do if they are forced to include the API key in the config file.

For example, a user's `~/.jccli` file might look like this:

```
[DEFAULT]
  Output = yaml
  OtherSettings = some_value
  KeyFile = ~/.jc_api.key
```

and then they would have another file, `~/.jc_api.key` that would just contain the key in it.

Therefore, I think we can merge this PR, and talk about how to do a config file in a separate PR.

Does this make sense?

Reply:
so i’ll say this @cascadianblue, the requirement for this is that there is a configuration file jccli configurations. The config file can store secrets and non-secrets.  if you want to set this up where a user can configure a path reference to another file, go right ahead, i am not opposed to it.